File: 4. Sorting/insertionSortList.js
function insertionSortList(head) {
  var before = { next: null };

  while (head) {
      var prev = before;

      // find prev
      while (prev.next && prev.next.val < head.val) {
          prev = prev.next;
      }

      var next = head.next;
      head.next = prev.next;
      prev.next = head;
      head = next;
  }

  return before.next;
}

File: 4. Sorting/insertion_sort_list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def insertionSortList(self, head: ListNode) -> ListNode:
        before = ListNode(None, None)

        while head:
          prev = before
          print(prev)
          while prev.next and prev.next.val < head.val:
            prev = prev.next

          nextPivot = head.next
          head.next = prev.next
          prev.next = head
          head = nextPivot

        return before.next





